pkg/agent/handler/accessrequest.go

```diff
@@ -0,0 +1,188 @@
+package handler
+
+import (
+	"encoding/json"
+	"fmt"
+
+	agentcache "github.com/Axway/agent-sdk/pkg/agent/cache"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	mv1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/management/v1alpha1"
+	defs "github.com/Axway/agent-sdk/pkg/apic/definitions"
+	prov "github.com/Axway/agent-sdk/pkg/apic/provisioning"
+	"github.com/Axway/agent-sdk/pkg/util"
+	"github.com/Axway/agent-sdk/pkg/watchmanager/proto"
+)
+
+const (
+	provision     = "Provision"
+	deprovision   = "Deprovision"
+	statusErr     = "Error"
+	statusSuccess = "Success"
+	statusPending = "Pending"
+)
+
+type arProvisioner interface {
+	AccessRequestProvision(accessRequest prov.AccessRequest) (status prov.RequestStatus)
+	AccessRequestDeprovision(accessRequest prov.AccessRequest) (status prov.RequestStatus)
+}
+
+type accessRequestHandler struct {
+	prov   arProvisioner
+	cache  agentcache.Manager
+	client client
+}
+
+// NewAccessRequestHandler creates a Handler for Access Requests
+func NewAccessRequestHandler(prov arProvisioner, cache agentcache.Manager, client client) Handler {
+	return &accessRequestHandler{
+		prov:   prov,
+		cache:  cache,
+		client: client,
+	}
+}
+
+func (h *accessRequestHandler) Handle(action proto.Event_Type, _ *proto.EventMeta, resource *v1.ResourceInstance) error {
+	if resource.Kind != mv1.AccessRequestGVK().Kind {
```

Comment:
We probably need a check to allow processing if provisioner interface is set(will be set only for DiscoveryAgent type)

Reply:
If there is no processor for a TA, then what will we be doing in here?